src/wireviz/wv_dataclasses.py

```diff
@@ -0,0 +1,815 @@
+            # Default True if any dimension specified unless self.scale also is specified.
+            self.fixedsize = (self.width or self.height) and self.scale is None
+
+        if self.scale is None:
+            if not self.width and not self.height:
+                self.scale = "false"
+            elif self.width and self.height:
+                self.scale = "both"
+            else:
+                self.scale = "true"  # When only one dimension is specified.
+
+        if self.fixedsize:
+            # If only one dimension is specified, compute the other
+            # because Graphviz requires both when fixedsize=True.
+            if self.height:
+                if not self.width:
+                    self.width = self.height * aspect_ratio(self.src)
+            else:
+                if self.width:
+                    self.height = self.width / aspect_ratio(self.src)
+
+
+@dataclass
+class PinClass:
+    index: int
+    id: str
+    label: str
+    color: MultiColor
+    parent: str  # designator of parent connector
+    _num_connections = 0  # incremented in Connector.connect()
+    _anonymous: bool = False  # true for pins on autogenerated connectors
+    _simple: bool = False  # true for simple connector
+
+    def __str__(self):
+        snippets = [  # use str() for each in case they are int or other non-str
+            str(self.parent) if not self._anonymous else "",
+            str(self.id) if not self._anonymous and not self._simple else "",
+            str(self.label) if self.label else "",
+        ]
+        return ":".join([snip for snip in snippets if snip != ""])
+
+
+@dataclass
+class Component:
+    category: Optional[str] = None  # currently only used by cables, to define bundles
+    type: Union[MultilineHypertext, List[MultilineHypertext]] = None
+    subtype: Union[MultilineHypertext, List[MultilineHypertext]] = None
+
+    # part number
+    partnumbers: PartNumberInfo = None  # filled by fill_partnumbers()
+    # the following are provided for user convenience and should not be accessed later.
+    # their contents are loaded into partnumbers during the child class __post_init__()
+    pn: str = None
+    manufacturer: str = None
+    mpn: str = None
+    supplier: str = None
+    spn: str = None
+    # BOM info
+    qty: NumberAndUnit = NumberAndUnit(1, None)
+    amount: Optional[NumberAndUnit] = None
+    sum_amounts_in_bom: bool = True
+    ignore_in_bom: bool = False
+    bom_id: Optional[str] = None  # to be filled after harness is built
+
+    def fill_partnumbers(self):
+        partnos = [self.pn, self.manufacturer, self.mpn, self.supplier, self.spn]
+        partnos = [remove_links(entry) for entry in partnos]
+        partnos = tuple(partnos)
+        self.partnumbers = PartNumberInfo(*partnos)
+
+    def parse_number_and_unit(
+        self,
+        inp: Optional[Union[NumberAndUnit, float, int, str]],
+        default_unit: Optional[str] = None,
+    ) -> Optional[NumberAndUnit]:
```

Comment:
Why is this a method in `class Component` and not a global function? The `self` parameter is not used.

Reply:
No real reason. OK to move to `vw_utils.py` or wherever you see fit.

Reply:
Moved `parse_number_and_unit` and `NumberAndUnit` definition to `wv_utils.py`.